stl/inc/iterator

```diff
@@ -511,6 +511,313 @@ private:
     bool _Failed; // true if any stores have failed
     streambuf_type* _Strbuf; // the wrapped stream buffer
 };
+
+#ifdef __cpp_lib_concepts
+// CLASS counted_iterator
+template <input_or_output_iterator _Iter>
+class counted_iterator {
+public:
+    using iterator_type = _Iter;
+
+    // [counted.iter.const]
+    constexpr counted_iterator() = default;
+    constexpr counted_iterator(_Iter _Right, const iter_difference_t<_Iter> _Diff) noexcept(
+        is_nothrow_move_constructible_v<_Iter>) // strengthened
+        : _Current(_STD move(_Right)), _Length(_Diff) {
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(_Diff >= 0, "counted_iterator requires positive length n");

```

Comment:
I believe that "positive" and "negative" strictly exclude zero. Since you're permitting zero here, this should say "non-negative":

```suggestion
        _STL_VERIFY(_Diff >= 0, "counted_iterator requires non-negative length n");
```

Reply:
true